client/my-sites/migrate/section-migrate.jsx

```diff
@@ -76,11 +87,35 @@ class SectionMigrate extends Component {
 			 * Note this migrationStatus is local, thus the setState vs setMigrationState.
 			 * Call to updateFromAPI will update both local and non-local state.
 			 */
-			this.setState( { migrationStatus: 'inactive', errorMessage: '' }, this.updateFromAPI );
+			this.setState(
+				{
+					migrationStatus: 'inactive',
+					errorMessage: '',
+				},
+				this.updateFromAPI
+			);
 		} );
 	};
 
 	setMigrationState = state => {
+		// Avoids a response from the migration-status endpoint
+		// accidentally resetting the state after the migrate/from endpoint
+		// has returned an error
+		if ( 'error' === this.state.migrationStatus ) {
+			return;
+		}
+
+		// Avoids a response from the migration-status endpoint
```

Comment:
Even with the comment, this seems confusing to me.

Reply:
Thanks, I've revised both comments to hopefully give a better idea of what we're doing.